decidim-admin/app/controllers/concerns/decidim/admin/filterable.rb

```diff
@@ -36,10 +39,64 @@ def query
 
         private
 
+        def check_admin_session_filters
+          if (current_filters = ransack_params).present?
+            admin_session_filters = session["admin_filters"] || {}
+            return if admin_session_filters[filter_prefix_key] == current_filters
+
+            current_filters = {} if current_filters[:reset_filters] == "true"
+
+            admin_session_filters[filter_prefix_key] = current_filters
+            session["admin_filters"] = admin_session_filters
+
+            redirect_to url_for(query_params.merge(q: {})) if current_filters.blank?
+          else
+            @session_filter_params = {} unless session_filter_params.is_a?(Hash)
+            redirect_to url_for(query_params_with(session_filter_params)) if session_filter_params.present?
+          end
+        end
+
         def filtered_collection
           paginate(query.result)
         end
 
+        def session_filtered_collection
+          @session_filtered_collection ||= begin
+            query = base_query.ransack(session_filter_params, search_context: :admin, auth_object: current_user).result
+            # The limit reorders as pagination does
+            query.limit(query.count)
+          end
+        end
+
+        def adjacent_items(item)
```

Comment:
Could you add a comment here to explain the result of the query?

Reply:
Added